src/components/backend-ai-usersettings-general-list.ts

```diff
@@ -593,6 +638,93 @@ export default class BackendAiUsersettingsGeneralList extends BackendAIPage {
     this.shadowRoot.querySelector('#change-current-editor-dialog').show();
   }
 
+  _openSSHKeypairManagementDialog() {
+    this.shadowRoot.querySelector('#ssh-keypair-management-dialog').show();
+  }
+
+  _openSSHKeypairGenerationDialog() {
+    this.shadowRoot.querySelector('#generate-ssh-keypair-dialog').show();
+  }
+
+  /**
+   * Fetch Existing Public key from Server.
+   * */
+  async _openSSHKeypairRefreshDialog() {
+    globalThis.backendaiclient.fetchSSHKeypair().then((resp) => {
+      const dialog = this.shadowRoot.querySelector('#ssh-keypair-management-dialog');
+      dialog.querySelector('#current-ssh-public-key').value = resp.ssh_public_key;
+      dialog.show();
+    });
+  }
+
+  _openSSHKeypairClearDialog() {
+    this.shadowRoot.querySelector('#clear-ssh-keypair-dialog').show();
+  }
+
+  _hideSSHKeypairGenerationDialog() {
+    this.shadowRoot.querySelector('#generate-ssh-keypair-dialog').hide();
+    const updatedSSHPublicKey: string = this.shadowRoot.querySelector('#ssh-public-key').value;
+    if (updatedSSHPublicKey !== "") {
+      const dialog = this.shadowRoot.querySelector('#ssh-keypair-management-dialog');
+      dialog.querySelector('#current-ssh-public-key').value = updatedSSHPublicKey;
+    }
+  }
+
+  _hideSSHKeypairDialog() {
+    this.shadowRoot.querySelector('#ssh-keypair-management-dialog').hide();
+  }
+
+  _hideSSHKeypairClearDialog() {
+    this.shadowRoot.querySelector('#clear-ssh-keypair-dialog').hide();
+  }
+
+  /**
+   * Fetch Randomly refreshed keypair generated from server.
+   * */
+  async _refreshSSHKeypair() {
+    const p = globalThis.backendaiclient.refreshSSHKeypair();
+    p.then((resp) => {
+      const sshKeyDialog = this.shadowRoot.querySelector('#generate-ssh-keypair-dialog');
+      sshKeyDialog.querySelector('#ssh-public-key').value = resp.ssh_public_key;
+      sshKeyDialog.querySelector('#ssh-private-key').value = resp.ssh_private_key;
+    });
+  }
+
+  /**
+   *  Generate SSH Keypair By Selected method (Custom or Random)
+   */
+  _generateSSHKeypair() {
+    const generateMethodElement = this.shadowRoot.querySelector('#keypair-generation-dropdown');
+    if (generateMethodElement.selected !== null) {
+      const generateMethod = generateMethodElement.value;
+      switch (generateMethod) {
+        case _text("usersettings.Random") : {
+          this._refreshSSHKeypair();
+          this.shadowRoot.querySelector('#ssh-public-key').readOnly = true;
+          this.shadowRoot.querySelector('#ssh-private-key').readOnly = true;
+          break;
+        }
+        case _text("usersettings.Custom") : {
+          /**
+           * TO DO : Custom SSH Keypair registration should be applied here.
+           */
```

Comment:
If this is what you asked to me in Teams, it will be good if you implement this feature.